config/sample-idex.cfg

```diff
@@ -44,13 +50,21 @@ gcode:
 
 # Activate the primary extruder
 [gcode_macro T0]
+variable_offset_applied: 0
 gcode:
-    PARK_{printer.toolhead.extruder}
+    {% set svv = printer.save_variables.variables %}
```

Comment:
I can see how saved variables could be useful, but this is complicating the understanding of what is required for setup.
It should be kept quite simple I think.
Possibly split it into basic, and advanced?

It's a flaw in my example macros as well. You really need succinct examples for basic docs, and everything needs to be obvious or well explained.

Reply:
I can add comments but I think this is the correct way to go here. Every commercial IDEX printer has a way to adjust the offset of the secondary extruder at runtime and the only way to do that and have the change persist across restarts is to use save variables. I also don't know how it's really complicating things at all. I think having a basic and advanced example would (1) just make the whole thing more complicated and (2) be unnecessary because I think everything else about IDEX printers is more complex anyway.

Reply:
It's a valid thing that basically every IDEX does things in a certain way, but remember if the macro is scary long, people are just going to copy it and not read it.
What are the purposes of this PR? To deliver a perfect solution or give people enough information to help themselves?

Reply:
> What are the purposes of this PR? To deliver a perfect solution or give people enough information to help themselves? 

Great question. My hope is to have enough documentation so that someone can *easily* setup an IDEX printer with Klipper. 

Now, defining "easily" is a complex thing to do for something that is inherently difficult. I would rather have a macro that works and that people won't read than a macro like the current one. The current T0 and T1 macros only have a Y offset, and require a whole restart just to update the x and y alignment. There is no mention of Z alignment. 

Basically, `if the macro is scary long, people are just going to copy it and not read it` yes, but if the macro doesn't work or is difficult to use then they aren't gonna use it at all. I'd rather them use it and then understand it later as opposed to spend 2 months gathering information from around the internet and experimenting and wasting a lot of pla like I did.

Reply:
Fair point. So you're aiming for a solution that will work sanely out the box.

I do think that it would be worth it to split it in 2 parts (PR's):
1. Get the docs scaffold in re IDEX
2. Work on the adding stuff to the scaffold.

Because that allows other people, e.g. like me, to help.

Reply:
Every time people mention splitting PRs I cringe because I have no idea how to split a branch without just branching and reverting different changes which seems dumb. 

I'll accept pull requests against my branch if you want... 

I think this conversation branch is getting away from where it started however which is the Save Variables element.

Reply:
@grigi Alright, a person on the internet is admitting they were wrong...

I have macros working but they still aren't perfect. I'm looking at things like z hop when changing extruders and stuff for reducing stringing but that is really getting nitty gritty and I'm probably not going to have a perfect printer for a couple months. 

I'm going to clean this up to the point you described above and I think we'll be ready for final review

Reply:
🤣  No worries 
Cura does hop/retract/temp/fan for you,  only need to park other toolhead on Tx.
Prusaslicer does nearly nothing, so for now I'm just using Cura 🤷 

Time is always a rarity

Reply:
I was having retraction quality problems with Cura :( I was able to get retraction to work with the same speed and distance in every slicer but Cura for some reason. I like the plugins that are available for cura, I like the interface for the most part, and I like the theory behind some of their decisions in structuring profiles but in the end tuning print quality was taking more time than just learning how to use superslicer